docker/server/nginx/etc/nginx/conf.d/universe.marathon.mesos.conf

```diff
@@ -48,15 +54,44 @@ server {
 
   location = /repo {
     types {
-      "application/vnd.dcos.universe.repo+json;charset=utf-8;version=v3" json;
       application/zip zip;
     }
 
-    if ($serve_json = true) {
-      rewrite ^/repo$ /repo-up-to-$dcos_release_version.json break;
+    if ($universe_version = v3) {
+      rewrite ^/repo$ /v3;
+    }
+
+    if ($universe_version = v4) {
+      rewrite ^/repo$ /v4;
     }
 
     rewrite ^/repo$ /repo-up-to-$dcos_release_version.zip break;
   }
 
+  location = /v3 {
+    types {
+      "application/vnd.dcos.universe.repo+json;charset=utf-8;version=v3" json;
+    }
+    
+    if ($dcos_release_version = "1.10") {
+      rewrite ^/v3$ /repo-empty-v3.json break;
+    }
```

Comment:
Ok, now that I see how this works, I changed my mind. Let's not have a special case for 1.10; we'll just let nginx rewrite the path, and if the file doesn't exist, it should automatically generate a 404. That will be easier to maintain because to change it, all we need to do is start generating the missing file.